homeworks/zsoltcsengeri/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,196 @@
+            max_distance = self.fuel_level / 0.1  # Max distance with remaining fuel
+            self.mileage += max_distance
+            self.fuel_level = 0  # Fuel is completely used up
+            print(
+                f"Not enough fuel for the full trip. {self.brand} {self.model} drove {max_distance:.2f} miles and ran out of fuel."
+            )
+
+    # Refuel method to fill up the tank with a given amount of fuel
+
+    def refuel(self, amount):
+        """
+        Refuel the car by a specified amount.
+        :param amount: Amount of fuel to add.
+        - If adding the fuel exceeds 100%, sets fuel level to 100% and notifies user of excess.
+        - Otherwise, adds the specified amount to the fuel level.
+        """
+        if amount < 0:
+            raise ValueError(
+                "Amount of fuel cannot be negative"
+            )  # Raise ValueError in case of a negative number
+
+        if self.fuel_level + amount > 100:
+            excess = self.fuel_level + amount - 100
+            self.fuel_level = 100  # Set fuel level to full
+            print(
+                f"The tank of {self.brand} {self.model} is now full! Could not add {excess}% as it exceeds the tank's capacity."
+            )
+        else:
+            self.fuel_level += amount
+            print(
+                f"The fuel level of {self.brand} {self.model} is now at {self.fuel_level}%."
+            )
+
+    def __str__(self):
+        return f"{self.brand} {self.model} {self.year} {self.mileage} {self.fuel_level}"
+
+
+# Example usage:
+
+car1 = Car(brand="Jaguar", model="E Pace", year=2018)
+print(car1)
+# Display initial car info
+car1.get_info()
+
+# Drive a certain distance
+# This will raise a ValueError, which we catch and handle here
+try:
+    car1.drive(-20)
+except ValueError as e:
+    # Print the error message to inform the user of invalid input
+    print(e)
+
+# Attempt to refuel the car
+# This will raise a ValueError, which we catch and handle here
+try:
+    car1.refuel(-5)
+except ValueError as e:
+    # Print the error message to inform the user of invalid input
+    print(e)
+
+"""
+*Create a Fleet class to manage Car objects:
+
+The class should have a list that contains the cars.
+Include methods to add and remove Car objects to/from the fleet.
+Include a method that aggregates the total mileage of all cars in the fleet.
+Create a few Car objects, add them to the fleet, perform some operations (driving, refueling), 
+and display the state of the cars and the summary data of the fleet.
+"""
+
+
+class Fleet:
+
+    def __init__(self, name: str):
+        self.name = name
+        self.cars = []
+
+    def add_car(self, car: Car) -> bool:
+        self.cars.append(car)  # Add a car to the fleet
+        return True
+
+    def list_cars(self) -> None:
+        print(f"Available cars in {self.name}: ")
+        for car in self.cars:
+            print(car)
+
+    def remove_car(self, brand: str) -> None:
```

Comment:
Ha csak a brand alapján törölsz, akkor ha pl több Ford típusú autó van a flottában, minden törlődni fog.
Szuper lenne ha magát az object-et törölnéd amit átadsz:

```
def remove_car(self, car:Car):
    if car in self.cars:
        self.cars.remove(car)
```

Így a kód is egyszerűbb, és az objektum memory address-e alapján töröl, ami egy egyedi azonosító.